// Copyright 2023 The Fleetbench Authors
//
// Licensed under the Apache License, Version 2.0 (the "License" );
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an " AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package fleetbench.rpc;

message DistributionArgs {
  optional string distribution = 1;

  optional int32 constant_value = 2 [default = 0];

  optional int32 uniform_min = 3 [default = 0];
  optional int32 uniform_max = 4 [default = 1];

  optional double exponential_lambda = 5 [default = 1];
  optional int32 exponential_max = 6 [default = 1000];

  optional double gaussian_mean = 7 [default = 0];
  optional double gaussian_stdev = 8 [default = 1];
  optional int32 gaussian_max = 9 [default = 1000];

  optional int32 poisson_mean = 10 [default = 1];
  optional int32 poisson_max = 11 [default = 1000];

  repeated int32 empirical = 12 [packed = true];
}
